src/pds_doi_service/core/outputs/datacite/datacite_web_client.py

```diff
@@ -147,19 +151,52 @@ def query_doi(self, query, url=None, username=None, password=None, content_type=
         logger.debug("query_string: %s", query_string)
         logger.debug("url: %s", url)
 
-        datacite_response = requests.request(
-            WEB_METHOD_GET, url=url, auth=auth, headers=headers, params={"query": query_string}
-        )
+        pages_returned = 0
 
         try:
+            # Submit the request, specifying that we would like up to 1000
+            # results returned for each page
+            datacite_response = requests.request(
+                WEB_METHOD_GET, url=url, auth=auth, headers=headers,
+                params={"query": query_string, "page[cursor]": 1, "page[size]": 1000}
+            )
+
             datacite_response.raise_for_status()
+
+            pages_returned += 1
+
+            # Parse the immediate result to see if we have any more pages to fetch
+            result = json.loads(datacite_response.text)
+
+            # Append current results to full set returned
+            data.extend(result['data'])
+
+            total_pages = result['meta']['totalPages']
+
+            # If necessary, request next page using the URL provided by DataCite
+            while pages_returned < total_pages:
+                url = result['links']['next']
+
+                datacite_response = requests.request(
+                    WEB_METHOD_GET, url=url, auth=auth, headers=headers
+                )
+
+                datacite_response.raise_for_status()
+
+                pages_returned += 1
+
+                # Append current results to full set returned
+                result = json.loads(datacite_response.text)
+                data.extend(result['data'])
```

Comment:
Will there ever be a concern that `data` grows overly large if, say, the `totalPages` is also overly large? Should we be using a generator style pattern here to compensate instead of gathering and returning the entire kit & kaboodle?

Reply:
I would say this is not *currently* a concern based on two things: 
1. Based on the performance I observed importing all 1300+ records currently available on DataCite (which is only about a page and half):

```
INFO __main__:main DOI import complete in 23.26 seconds.
INFO __main__:main Num records found: 1390
INFO __main__:main Num records processed: 1390
INFO __main__:main Num records written: 1390
INFO __main__:main Num records skipped: 0
``` 

2.  This query function is only used right now for the local database importation script, so for the time being we shouldn't have to worry about an end-user of the service/UI running into some sort of performance bottleneck because we slurp all available pages.